src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1340,6 +1340,21 @@ public void SetBadRequestState(BadHttpRequestException ex)
             _keepAlive = false;
         }
 
+        // Normally this would have been rejected, but the developer opted into allowing the bad behavior.
+        public void ReportAllowedBadRequest(BadHttpRequestException ex)
+        {
+            Log.ConnectionBadRequest(ConnectionId, ex);
+            _requestRejectedException = ex;
```

Comment:
```suggestion
            // Set this so DiagnosticSource listeners can observe the exception via the IBadRequestExceptionFeature.
            // Make sure to unset before exiting the method so the request doesn't actually get rejected.
            // This means the exception will not be visible to normal middleware.
            _requestRejectedException = ex;
```

Alternatively, we could create another field for `_allowedBadRequestException` or something so it's visible from middleware too. I don't think that's necessary though.